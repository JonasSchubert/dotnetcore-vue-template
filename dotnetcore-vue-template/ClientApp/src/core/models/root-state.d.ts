import { ExamplesState } from '@/examples/models';
import { BaseState } from "@/libs/core/models";
import { I18nState } from "@/libs/i18n/models";
import { LocalStorageState } from "@/libs/local-storage/models";

export interface RootState extends BaseState {
  examples?: ExamplesState;
  i18n?: I18nState;
  localStorage?: LocalStorageState;
}
