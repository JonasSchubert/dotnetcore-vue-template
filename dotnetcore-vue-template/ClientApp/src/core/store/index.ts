import Vue from 'vue';
import Vuex, { Store, StoreOptions } from 'vuex';
import { createI18nModule } from '@/core/i18n/module-helper';
import { RootState } from '@/core/models';
import { createModule as createExamplesModule } from '@/examples/store';
import { ModuleType as ExamplesModule } from '@/examples/store/types';
import { ModuleType as LocalStorageModule } from "@/libs/local-storage/store/types";
import { createLocalStorageModule } from "@/libs/local-storage/store";
import { createGetters } from './getters';
import { createMutations } from './mutations';
import { createState } from './state';

Vue.use(Vuex);

export const createStore = (vuetify: any): Store<RootState> => {
  const store: StoreOptions<RootState> = {
    getters: createGetters(),
    modules: {
      [ExamplesModule]: createExamplesModule(),
      i18n: createI18nModule(),
      [LocalStorageModule]: createLocalStorageModule<RootState>('dotnetcore-vue-template', vuetify)
    },
    mutations: createMutations(),
    state: createState(),
    strict: process.env.NODE_ENV !== 'production'
  };

  return new Store<RootState>(store);
};
