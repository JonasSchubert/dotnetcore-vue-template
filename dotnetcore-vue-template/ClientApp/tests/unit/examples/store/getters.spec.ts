import { RootState } from '@/core/models';
import { Example, ExamplesState } from '@/examples/models';
import { createGetters } from '@/examples/store/getters';
import { GetterTypes } from '@/examples/store/types';

describe('examples/getters', () => {
  let state: ExamplesState;
  const getters = createGetters();

  beforeEach(() => {
    state = {
      error: {
        message: 'message',
        name: 'name'
      },
      isLoading: true,
      list: [
        { id: '1' } as Example
      ]
    };
  });

  describe(`${GetterTypes.error}`, () => {
    test('should return error as expected', () => {
      // Arrange && Act
      const actual = getters[GetterTypes.error](state, getters, {} as RootState, {});

      // Assert
      expect(actual).toMatchSnapshot({
        message: 'message',
        name: 'name'
      });
    });
  });

  describe(`${GetterTypes.isLoading}`, () => {
    test('should return isLoading as expected', () => {
      // Arrange && Act
      const actual = getters[GetterTypes.isLoading](state, getters, {} as RootState, {});

      // Assert
      expect(actual).toBeTruthy();
    });
  });

  describe(`${GetterTypes.list}`, () => {
    test('should return list as expected', () => {
      // Arrange && Act
      const actual = getters[GetterTypes.list](state, getters, {} as RootState, {});

      // Assert
      expect(actual).toMatchSnapshot([
        { id: '1' } as Example
      ]);
    });
  });
});
