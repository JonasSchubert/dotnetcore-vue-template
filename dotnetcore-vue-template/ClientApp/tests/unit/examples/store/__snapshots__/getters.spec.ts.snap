// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples/getters error should return error as expected 1`] = `
Object {
  "message": "message",
  "name": "name",
}
`;

exports[`examples/getters list should return list as expected 1`] = `
Object {
  "0": Object {
    "id": "1",
  },
}
`;
